.github/workflows/builds.yml

```diff
@@ -0,0 +1,54 @@
+name: builds
+
+on:
+  release:
+    types: [created]
+  push:
+    branches:
+      - master
+      - main
+  pull_request:
+
+jobs:
+  docker-oss:
+    runs-on: ubuntu-latest
+    steps:
+      -
+        name: Checkout
+        uses: actions/checkout@v2
+      -
+        name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: |
+            name=rudderlabs/develop-rudder-server
+            name=rudderlabs/rudder-server,enable=${{ github.ref == format('refs/heads/{0}', 'master') || github.event_name == 'release' }}
+          tags: |
+            type=ref,event=branch
+            type=raw,value=latest,enable={{is_default_branch}}
+            type=raw,value=${{ github.head_ref }}
+            type=semver,pattern={{version}}
+            type=semver,pattern={{major}}.{{minor}}
+            type=semver,pattern={{major}}
+
+      -
+        name: Login to DockerHub
+        uses: docker/login-action@v2
+        with:
+          username: rudderlabs
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      -
+        name: Build and push
+        uses: docker/build-push-action@v3
```

Comment:
If you want to speed this up you could use the `buildx` cache like I did [here](https://github.com/rudderlabs/rudder-scheduler/blob/29632ab300b79a1f488cdb08a43231540d58a7de/.github/workflows/tests.yaml#L37) (see `name: Cache Docker layers` and then the `cache-from` and `cache-to` in the `docker/build-push-action@v3`).